libraries/utilities.js

```diff
@@ -875,4 +877,23 @@ const getVideoDir = function (objectsPath, identityFolderName, isMobile, objectN
 
     return videoDir;
 }
-exports.getVideoDir = getVideoDir;
\ No newline at end of file
+exports.getVideoDir = getVideoDir;
+
+exports.setupNodeTypes = function(nodeTypeModules) {
+    nodeTypes = nodeTypeModules;
+};
+
+const createNode = function(name, type) {
```

Comment:
Why not have this be the Node constructor? It looks like every use of `new Node` has a name and type assigned at some point following. Is it for the private/public data?

Reply:
Oh, that actually makes a lot of sense!

Reply:
Essentially we just need the public and private data to be copied from the definition of the node type, so I think that's a reasonable modification of the constructor